### PR TITLE
prop_conv_solvert::get_bool now returns optional

### PR DESCRIPTION
**Pull request by kroening**
- [X] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: 1c678c7ef0c267389b6f3bf0ce0bf60deda64f60).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/107532945
Status will be re-evaluated on next push.
Common spurious failures include: the cbmc commit has disappeared in the mean time (e.g. in a force-push); the author is not in the list of contributors (e.g. first-time contributors); compatibility was already broken by an earlier merge.

File: src/solvers/prop/prop_conv_solver.h [hunk 1]
```diff
@@ -112,7 +112,7 @@ class prop_conv_solvert : public prop_convt, public solver_resource_limitst
   bool post_processing_done = false;
 
   // get a _boolean_ value from counterexample if not valid
-  virtual bool get_bool(const exprt &expr, tvt &value) const;
+  virtual optionalt<bool> get_bool(const exprt &expr) const;
```
Comment (tautschnig) [inline src/solvers/prop/prop_conv_solver.h:117]: Maybe this method can actually receive proper documentation? It doesn't at all discuss the error case, and the "if not valid" lacks a subject.

**Review by tautschnig — APPROVED** [review]
Seems ok, but wouldn't the use of `tvt` as the return type be very natural here? The result quite literally is true, false, or unknown!

**Review by peterschrammel — APPROVED** [review]

File: src/solvers/prop/prop_conv_solver.h (hunk 1 above)
Comment (kroening) [reply src/solvers/prop/prop_conv_solver.h:117]: fixed

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 8dbfae17486f9903edd72d1748543d0e6151f2ae).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/108218296

**Pull request merged**